Convert sparse object to sparse tensor before evaluating model.

Abdul-Moeed opened the pull request:
Closes #7

davidsebfischer commented:
who only in eval/test and not in train? could also make sense to make it a further option to return sparse tensors in the generator, in principle this could also happen / be useful on non backed data.

Abdul-Moeed commented:
Both suggestions sound good! I will implement them as soon as I get some free time. We can merge this after these changes are in the PR.

Abdul-Moeed commented:
I have refactored `_get_dataset()` quite a lot, I will run some tests to ensure everything works as expected.
We now always get a dataset generator from this function. I have also modified `evaluate()` and `evaluate_any()` to take this change into account.
Regarding sparse tensors: Combining dataset generators and sparse tensors is not very intuitive, see this issue for more. Currently the function returns dense tensors, but if the performance gain is worth it, maybe we can do something like [this](https://github.com/tensorflow/tensorflow/issues/16689).
